staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/apiserver.go

```diff
@@ -244,6 +248,55 @@ func (c completedConfig) New(delegationTarget genericapiserver.DelegationTarget)
 		go finalizingController.Run(5, context.StopCh)
 		return nil
 	})
+	s.GenericAPIServer.AddPostStartHookOrDie("crd-discovery-available", func(context genericapiserver.PostStartHookContext) error {
+		return wait.PollImmediateUntil(100*time.Millisecond, func() (bool, error) {
+			// only check if we have a valid list for a given resourceversion
+			if !s.Informers.Apiextensions().V1().CustomResourceDefinitions().Informer().HasSynced() {
```

Comment:
accessing `s.Informers.Apiextensions().V1().CustomResourceDefinitions().Informer()` potentially after we've started informers (line 231) is fragile, and counts on someone else having also accessed the same informer